esmvalcore/_citation.py

```diff
@@ -0,0 +1,218 @@
+    'Technical overview}\n'
+    '}\n'
+)
+
+
+def _write_citation_file(filename, provenance):
+    """
+    Write citation information provided by the recorded provenance.
+
+    Recipe and cmip6 data references are saved into one bibtex file.
+    cmip6 data references are provided by CMIP6 data citation service.
+    each cmip6 data reference has a json link. In the case of internet
+    connection, cmip6 data references are saved into a bibtex file.
+    Otherwise, cmip6 data reference links are saved into a text file.
+    """
+    product_name = os.path.splitext(filename)[0]
+    info_urls = []
+    json_urls = []
+    product_tags = []
+    for item in provenance.records:
+        reference_attr = item.get_attribute('attribute:references')
+        # get cmip6 citation info
+        value = item.get_attribute('attribute:mip_era')
+        if 'CMIP6' in value:
+            url_prefix = _make_url_prefix(item.attributes)
+            info_urls.append(_make_info_url(url_prefix))
+            json_urls.append(_make_json_url(url_prefix))
+        if reference_attr:
+            # get recipe citation tags
+            if item.identifier.namespace.prefix == 'recipe':
+                product_tags.extend(reference_attr)
+            # get diagnostics citation tags
+            elif item.get_attribute('attribute:script_file'):
+                product_tags.extend(reference_attr)
+            else:
+                info_urls += list(reference_attr)
+
+    _save_citation_info(product_name, product_tags, json_urls, info_urls)
+
+
+def _save_citation_info(product_name, product_tags, json_urls, info_urls):
+    citation_entries = [ESMVALTOOL_PAPER]
+
+    # save CMIP6 url_info, if any
+    # save any refrences info that is not related to recipe or diagnostics
+    title = [
+        "Some citation information are found, "
+        "which are not mentioned in the recipe or diagnostic."
+    ]
+    if info_urls:
+        with open(f'{product_name}_data_citation_info.txt', 'w') as file:
+            file.write('\n'.join(title + list(set(info_urls))))
+
+    # convert json_urls to bibtex entries
+    for json_url in json_urls:
+        cmip_citation = _collect_cmip_citation(json_url)
+        if cmip_citation:
+            citation_entries.append(cmip_citation)
+
+    # convert tags to bibtex entries
+    if REFERENCES_PATH and product_tags:
+        tags = _extract_tags(product_tags)
+        for tag in tags:
+            citation_entries.append(_collect_bibtex_citation(tag))
+
+    with open(f'{product_name}_citation.bibtex', 'w') as file:
+        file.write('\n'.join(citation_entries))
+
+
+def _extract_tags(tags):
+    """
+    Extract tags that are recorded by provenance as
+    for example, "['acknow_project', 'acknow_author']".
+    """
+    pattern = re.compile(r'\w+')
+    return list(set(pattern.findall(str(tags))))
+
+
+def _get_response(url):
+    """Return information from CMIP6 Data Citation service in json format."""
+    json_data = False
```

Comment:
```suggestion
    json_data = None
```
or `{}` would be better, because then the function has a consistent return type